crates/consensus/src/transaction/eip4844/builder.rs

```diff
@@ -0,0 +1,390 @@
+#[cfg(not(feature = "kzg"))]
+use alloy_eips::eip4844::Blob;
+#[cfg(feature = "kzg")]
+use c_kzg::{Blob, KzgCommitment, KzgProof, KzgSettings};
+
+use alloy_eips::eip4844::{BYTES_PER_BLOB, FIELD_ELEMENTS_PER_BLOB};
+
+use super::utils::WholeFe;
+
+/// A builder for creating a [`BlobTransactionSidecar`].
+///
+/// [`BlobTransactionSidecar`]: crate::BlobTransactionSidecar
+#[derive(Debug, Clone)]
+pub struct PartialSidecar {
+    /// The blobs in the sidecar.
+    blobs: Vec<Blob>,
+    /// The number of field elements that we have ingested, total.
+    fe: usize,
+}
+
+impl Default for PartialSidecar {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl PartialSidecar {
+    /// Create a new builder.
+    pub fn new() -> Self {
+        // NB: vecs default to 100 capacity. Blobs are large. We don't want
+        // to allocate 100 blobs if we don't need them.
+        Self::with_capacity(2)
+    }
+
+    /// Create a new builder with a given capacity.
+    pub fn with_capacity(capacity: usize) -> Self {
+        let mut blobs = Vec::with_capacity(capacity);
+        blobs.push(Blob::new([0u8; BYTES_PER_BLOB]));
```

Comment:
this also adds a new blob, 
looking at the other function, I think this is intended

Reply:
yeah, the intention is to never allow the blobs vec to be empty

Reply:
i've improved instantiation quite a bit now :)